src/map/ai/controllers/automaton_controller.cpp

```diff
@@ -1333,8 +1333,11 @@ bool CAutomatonController::TryTPMove()
 
 bool CAutomatonController::TryRangedAttack() // TODO: Find the animation for its ranged attack
 {
-    if (m_rangedCooldown > 0s && m_Tick > m_LastRangedTime + (m_rangedCooldown - std::chrono::seconds(PAutomaton->getMod(Mod::SNAP_SHOT))))
-        return MobSkill(PTarget->targid, m_RangedAbility);
+    if (PAutomaton->getFrame() == FRAME_SHARPSHOT) {
+        if (m_rangedCooldown > 0s && m_Tick > m_LastRangedTime + (m_rangedCooldown - std::chrono::seconds(PAutomaton->getMod(Mod::SNAP_SHOT))))
+            return MobSkill(PTarget->targid, m_RangedAbility);
+        return false;
```

Comment:
Since we're returning false outside of this parent if check, we can combine the frame check into the same check as the cooldown check, and remove the inner `return false`.

Reply:
Doh cs101 change made ;)